tests/platform_tests/test_auto_negotiation.py

```diff
@@ -453,3 +453,33 @@ def get_cable_supported_speeds(cls, duthost, dut_port_name):
         speeds =  [speed[:-1] + '000' for speed in speeds_str.split(',')]
         cls.supported_speeds[(duthost, dut_port_name)] = speeds
         return speeds
+
+    @classmethod
+    def get_port_index_for_mlxlink(cls, duthost, dut_port_name):
```

Comment:
can we use this API to get the port index for a logical port https://github.com/Azure/sonic-mgmt/blob/master/tests/common/platform/interface_utils.py#L135 instead of using mellanox specific?